eth2util/rlp/rlp.go

```diff
@@ -113,7 +113,7 @@ func decodeLength(item []byte) (offset int, length int, err error) {
 			return 0, 0, err
 		} else if resp < 0 {
 			return 0, 0, errors.New("negative length")
-		} else if offset > offset+resp {
+		} else if offset+resp > len(item) {
```

Comment:
not sure about this

Reply:
isn't `offset > offset+resp` same as `resp < 0`?

Reply:
you can move the input length check into the `decodeLength`, but then you should remove it from the caller side

Reply:
@xenowits why is this check better than `offset > offset+resp`?

Reply:
@gsora there was a confusion around overflowing: `overflow of length of input` (1) vs `overflow of int64` (2). I'm reverting this check since the caller checks (1) and `decodeLength` checks (2).